internal/platform/web/client.go

```diff
@@ -0,0 +1,28 @@
+package web
+
+import (
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	// DefaultTimeout is the default amount of time for a request to complete before it times out.
+	DefaultTimeout = 30 * time.Second
+)
+
+func NewHTTPClient() *http.Client {
```

Comment:
I don't like this for several reasons.

This is hiding cost for the sake of not repeating yourself. I look at the call and I don't know what the settings are. This is important. This function is not creating a new semantic where things are precise.

Also, do not construct values to variables using pointer semantics. The dialer should be a value and the & should be applied to line 23. The same on the construction of the client.

I would rather take this simplier form of construction the dialer and client and replace this code with the old code.

Don't feel that repeating yourself is wrong. This is code that comes from the standard library and it doesn't need to be abstracted. Plus, if one of the places this is used needs special settings, we lost that. Too much being hidden for no gain.

Reply:
Losing the setting is not good either.

https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/

Let's not do this at all.

Reply:
Ok, I try to extract to a factory method a initial implementation just for the sake of DRY but taking into account future customizations like:

https://github.com/tsenart/vegeta/blob/8c061dc2715c10b49d5f47fbaf07c3f4c3bce4c7/lib/attack.go

https://github.com/emacampolo/gomparator/blob/master/internal/platform/http/client.go

https://github.com/hashicorp/go-retryablehttp/blob/master/client.go

Reply:
I will rollback that change as well

Reply:
If we had special platform or busines logic, this would make sense. We are not. This is just basic transport and client construction. It doesn't need the level of abstraction. Just work with these types directly.